src/dmd/backend/elfobj.d

```diff
@@ -1635,8 +1634,9 @@ void Obj_compiler()
     comment_data = cast(Outbuffer*) calloc(1, Outbuffer.sizeof);
     assert(comment_data);
 
-
-    comment_data.write(.compiler,(.compiler).sizeof);
+    char[20 + 8 /* v2.0XX.X */] compiler = void;
+    sprintf(compiler.ptr, "\0Digital Mars C/C++ %s", global._version);
```

Comment:
1) `sprintf` is an overkill here. This would be `@safe` (though a bit longer)

```
char[20 + 8 /* v2.0XX.X */] compiler = void;
enum compilerHeader = "\0Digital Mars C/C++ %s";
enum n = compilerHeader.length;
compiler[0 .. n] = compilerHeader;
auto versionArr = &global._version[0 .. strlen(global._version)];
assert(versionArr.length < 16);
compiler[n .. n + versionArr.length] = versionArr;
```

2) The buffer needs to be a lot larger:

This is e.g. the DMD debug version I currently have:

```
v2.084.0-390-g08394cbae-dirty
```

(could be a bit bigger as it's `git describe --dirty`)